src/main/java/it/gov/pagopa/reward/controller/RewardTrxSynchronousApiControllerImpl.java

```diff
@@ -23,9 +23,19 @@ public RewardTrxSynchronousApiControllerImpl(RewardTrxSynchronousApiService rewa
     public Mono<SynchronousTransactionResponseDTO> previewTransaction(SynchronousTransactionRequestDTO trxPreviewRequest, String initiativeId) {
         log.info("[SYNC_PREVIEW_TRANSACTION] The user {} requests preview of a transaction", trxPreviewRequest.getUserId());
 
-        return PerformanceLogger.logTimingFinally("[SYNC_PREVIEW_TRANSACTION]",
+        return PerformanceLogger.logTimingFinally("SYNC_PREVIEW_TRANSACTION",
                 rewardTrxSynchronousService.previewTransaction(trxPreviewRequest, initiativeId)
                         .switchIfEmpty(Mono.error(new ClientExceptionNoBody(HttpStatus.NOT_FOUND,"NOTFOUND"))),
                 trxPreviewRequest.toString());
     }
+
+    @Override
+    public Mono<SynchronousTransactionResponseDTO> authorizeTransaction(SynchronousTransactionRequestDTO trxAuthorizeRequest, String initiativeId) {
+        log.info("[SYNC_AUTHORIZE_TRANSACTION] The user {} requests authorize transaction {}", trxAuthorizeRequest.getUserId(), trxAuthorizeRequest.getTransactionId());
+
+        return PerformanceLogger.logTimingFinally("SYNC_AUTHORIZE_TRANSACTION",
+                rewardTrxSynchronousService.authorizeTransaction(trxAuthorizeRequest, initiativeId)
+                        .switchIfEmpty(Mono.error(new ClientExceptionNoBody(HttpStatus.NOT_FOUND,"NOTFOUND"))),
```

Comment:
what is not found?
Please, make this changes also preview API
```suggestion
                        .switchIfEmpty(Mono.error(new ClientExceptionNoBody(HttpStatus.NOT_FOUND,"Cannot find initiative having id " + initiativeId))),
```